File: jvm-experiments-commons/src/main/java/kr/kth/commons/timeperiod/DayOfWeek.java
package kr.kth.commons.timeperiod;

import org.joda.time.DateTimeConstants;

/**
 * kr.kth.commons.timeperiod.DayOfWeek
 * User: sunghyouk.bae@gmail.com
 * Date: 12. 12. 27.
 */
public enum DayOfWeek {

    Monday(DateTimeConstants.MONDAY),
    Tuesday(DateTimeConstants.TUESDAY),
    Wednesday(DateTimeConstants.WEDNESDAY),
    Thursday(DateTimeConstants.THURSDAY),
    Friday(DateTimeConstants.FRIDAY),
    Saturday(DateTimeConstants.SATURDAY),
    Sunday(DateTimeConstants.SUNDAY);


    private final int day;

    DayOfWeek(int day) {
        this.day = day;
    }

    public int toInt() {
        return day;
    }

    public static DayOfWeek valueOf(int dayOfWeek) {

        int day = (dayOfWeek % TimeSpec.DaysPerWeek) + 1;

        switch (day) {
            case DateTimeConstants.MONDAY:
                return DayOfWeek.Monday;

            case DateTimeConstants.TUESDAY:
                return DayOfWeek.Tuesday;

            case DateTimeConstants.WEDNESDAY:
                return DayOfWeek.Wednesday;

            case DateTimeConstants.THURSDAY:
                return DayOfWeek.Thursday;

            case DateTimeConstants.FRIDAY:
                return DayOfWeek.Friday;

            case DateTimeConstants.SATURDAY:
                return DayOfWeek.Saturday;

            case DateTimeConstants.SUNDAY:
                return DayOfWeek.Sunday;

            default:
                throw new IllegalArgumentException("dayOfWeek 값이 잘못되었습니다. dayOfWeek=" + dayOfWeek);
        }
    }
}

File: jvm-experiments-timeperiod/src/main/scala/kr/kth/timeperiod/DayOfWeek.scala
package kr.kth.timeperiod

import org.joda.time.DateTimeConstants

/**
 * 한 주의 요일을 표현합니다.
 * User: sunghyouk.bae@gmail.com
 * Date: 13. 1. 13.
 */
object DayOfWeek extends Enumeration {

  type DayOfWeek = Value

  // NOTE: withName, apply 메소드를 사용하세요

  val Monday = Value(DateTimeConstants.MONDAY, "Monday")
  val Tuesday = Value(DateTimeConstants.TUESDAY, "Tuesday")
  val Wednesday = Value(DateTimeConstants.WEDNESDAY, "Wednesday")
  val Thursday = Value(DateTimeConstants.THURSDAY, "Thursday")
  val Friday = Value(DateTimeConstants.FRIDAY, "Friday")
  val Saturday = Value(DateTimeConstants.SATURDAY, "Saturday")
  val Sunday = Value(DateTimeConstants.SUNDAY, "Sunday")
}
